.travis.yml

```diff
@@ -14,19 +14,29 @@ services:
 # support for remote replicas is added.
 env:
   global:
+    - OS=ubuntu
+    - OS_VER=16.04
+    - MAKE_PKG=0
+    - PMDK_CC=gcc
+    - PMDK_CXX=g++
+    - REMOTE_TESTS=1
+    - EXTRA_CFLAGS=-DUSE_VALGRIND
+    - EXTRA_CXXFLAGS=-DUSE_VALGRIND
+    - LIBCPP_LIBDIR=/usr/local/libcxx/lib
+    - LIBCPP_INCDIR=/usr/local/libcxx/include/c++/v1
     - NDCTL_ENABLE=n
   matrix:
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=gcc PMDK_CXX=g++ OS=ubuntu OS_VER=16.04 TEST_BUILD=debug COVERAGE=1
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=gcc PMDK_CXX=g++ OS=ubuntu OS_VER=16.04 TEST_BUILD=debug
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=gcc PMDK_CXX=g++ OS=ubuntu OS_VER=16.04 TEST_BUILD=nondebug
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 LIBCPP_LIBDIR=/usr/local/libcxx/lib LIBCPP_INCDIR=/usr/local/libcxx/include/c++/v1 OS=ubuntu OS_VER=16.04 TEST_BUILD=debug
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 LIBCPP_LIBDIR=/usr/local/libcxx/lib LIBCPP_INCDIR=/usr/local/libcxx/include/c++/v1 OS=ubuntu OS_VER=16.04 TEST_BUILD=nondebug
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 LIBCPP_LIBDIR=/usr/local/libcxx/lib LIBCPP_INCDIR=/usr/local/libcxx/include/c++/v1 OS=fedora OS_VER=25 TEST_BUILD=debug
-    - MAKE_PKG=0 EXTRA_CFLAGS=-DUSE_VALGRIND EXTRA_CXXFLAGS=-DUSE_VALGRIND REMOTE_TESTS=1 PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 LIBCPP_LIBDIR=/usr/local/libcxx/lib LIBCPP_INCDIR=/usr/local/libcxx/include/c++/v1 OS=fedora OS_VER=25 TEST_BUILD=nondebug
-    - MAKE_PKG=1 PMDK_CC=gcc PMDK_CXX=g++ OS=ubuntu OS_VER=16.04 EXPERIMENTAL=y
-    - MAKE_PKG=1 PMDK_CC=clang PMDK_CXX=clang++ OS=ubuntu OS_VER=16.04 EXPERIMENTAL=y
-    - MAKE_PKG=1 PMDK_CC=gcc PMDK_CXX=g++ OS=fedora OS_VER=25 EXPERIMENTAL=y NDCTL_ENABLE=y
-    - COVERITY=1 REMOTE_TESTS=1 PMDK_CC=gcc PMDK_CXX=g++ OS=ubuntu OS_VER=16.04
+    - COVERAGE=1 TEST_BUILD=debug
+    - TEST_BUILD=debug
+    - TEST_BUILD=nondebug
+    - PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 TEST_BUILD=debug
+    - PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 TEST_BUILD=nondebug
+    - OS=fedora OS_VER=25 PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 TEST_BUILD=debug
+    - OS=fedora OS_VER=25 PMDK_CC=clang PMDK_CXX=clang++ USE_LLVM_LIBCPP=1 TEST_BUILD=nondebug
+    - MAKE_PKG=1 EXPERIMENTAL=y REMOTE_TESTS=0 EXTRA_CFLAGS= EXTRA_CXXFLAGS=
+    - MAKE_PKG=1 EXPERIMENTAL=y REMOTE_TESTS=0 PMDK_CC=clang PMDK_CXX=clang++ EXTRA_CFLAGS= EXTRA_CXXFLAGS=
```

Comment:
Why empty extra flags?

Reply:
To override defaults and compile w/o Valgrind.